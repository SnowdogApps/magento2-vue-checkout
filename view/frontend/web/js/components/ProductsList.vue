<template>
  <section
    v-if="step !== 'success'"
    class="products-list"
  >
    <h2>
      {{ $t('Product List') }}
    </h2>
    <ul class="products">
      <ProductsListItem
        v-for="product in products"
        :key="product.id"
        :product="product"
      />
    </ul>
  </section>
</template>

<script>
import ProductsListItem from './ProductsListItem.vue'

export default {
  components: {
    ProductsListItem
  },
  computed: {
    products () {
      return this.$store.state.config.totalsData.items
    },
    step () {
      return this.$store.state.step
    }
  }
}
</script>

<style lang="scss" scoped>
  .products-list {
    margin-top: 10px;
    padding: 10px;
    border: 1px solid #c2c2c2;
  }
</style>
