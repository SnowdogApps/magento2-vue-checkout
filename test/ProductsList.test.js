import Vue from 'vue'
import Vuei18n from 'vue-i18n'
import { mount, createLocalVue } from '@vue/test-utils'
import messages from '../view/frontend/web/js/data/translations-en_US.json'
import ProductsList from '../view/frontend/web/js/components/ProductsList.vue'

const localVue = createLocalVue()
localVue.use(Vuei18n)
const i18n = new Vuei18n({
  locale: 'en_US',
  messages
})

Vue.filter('currency', function (price) {
  const pattern = '$%s'
  price = parseFloat(price).toFixed(2)
  return pattern.replace('%s', price)
})

describe('ProductsList.test.js', () => {
  let wrapper

  beforeEach(() => {
    wrapper = mount(ProductsList, {
      computed: {
        products: () => [
          {
            base_discount_amount: '0.0000',
            base_price: '48.0000',
            base_price_incl_tax: '48.0000',
            base_row_total: '48.0000',
            base_row_total_incl_tax: '48.0000',
            base_tax_amount: '0.0000',
            discount_amount: '0.0000',
            discount_percent: '0.0000',
            item_id: '229',
            name: 'Hera Pullover Hoodie',
            options: '[{"value":"Blue","label":"Color"},{"value":"XS","label":"Size"}]',
            price: '48.0000',
            price_incl_tax: '48.0000',
            qty: 1,
            row_total: '48.0000',
            row_total_incl_tax: '48.0000',
            row_total_with_discount: '0.0000',
            tax_amount: '0.0000',
            tax_percent: '0.0000',
            weee_tax_applied: null,
            weee_tax_applied_amount: null
          }
        ],
        step: () => 'shipping'
      },
      localVue,
      i18n
    })
  })

  it('renders a $store.state products list return from computed', () => {
    expect(wrapper.find('.products').contains('.product')).toBe(true)
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
