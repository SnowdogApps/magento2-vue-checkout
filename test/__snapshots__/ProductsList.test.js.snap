// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList.test.js has the expected html structure 1`] = `
<section
  class="products-list"
>
  <h2>
    
    Product List
  
  </h2>
   
  <ul
    class="products"
  >
    <li
      class="product"
    >
      <p
        class="products__title"
      >
        
    Product name:
    
        <strong>
          
      Hera Pullover Hoodie
    
        </strong>
      </p>
       
      <span
        class="product__price"
      >
        
    Price:

    
        <strong>
          
      $48.00
    
        </strong>
      </span>
       
      <div
        class="products__handler"
      >
        <p
          class="products__information"
        >
          
      Product details:
    
        </p>
         
        <span
          class="products__detail"
        >
          
        Color:
        
          <strong>
            
          Blue
        
          </strong>
        </span>
        <span
          class="products__detail"
        >
          
        Size:
        
          <strong>
            
          XS
        
          </strong>
        </span>
         
        <span
          class="products__detail"
        >
          
      Qty:
      
          <strong>
            
        1
      
          </strong>
        </span>
         
        <!---->
      </div>
    </li>
  </ul>
</section>
`;
